modules/aerodyn/src/AeroDyn_IO.f90

```diff
@@ -1030,22 +1203,106 @@ logical function Failed()
+      character(len=*),           intent(in  )  :: varName     !< Variable being read
+      integer(IntKi),             intent(in   ) :: iLine       !< Line number
+      integer(IntKi),             intent(inout) :: errStat     !< Error status
+      character(ErrMsgLen),       intent(inout) :: errMsg      !< Error message
+      character(len=*), optional, intent(in  )  :: varNameSubs !< Substituted variable
+      legacyInputPresent = errStat == ErrID_None
+      if (legacyInputPresent) then
+         if (present(varNameSubs)) then
+            call LegacyWarning(trim(varName)//' has now been removed.'//NewLine//'    Use: '//trim(varNameSubs)//'.')
+         else
+            call LegacyWarning(trim(varName)//' has now been removed.')
+         endif
+      else
+         ! We are actually happy, this input should indeed not be present.
+      endif
+      ! We erase the error no matter what
+      errStat = ErrID_None
+      errMsg  = ''
+   end function legacyInputPresent
+   !-------------------------------------------------------------------------------------------------
+   logical function newInputMissing(varName, iLine, errStat, errMsg, varNameSubs)
+      character(len=*),           intent(in  )  :: varName     !< Variable being read
+      integer(IntKi),             intent(in   ) :: iLine       !< Line number
+      integer(IntKi),             intent(inout) :: errStat     !< Error status
+      character(ErrMsgLen),       intent(inout) :: errMsg      !< Error message
+      character(len=*), optional, intent(in  )  :: varNameSubs !< Substituted variable
+      newInputMissing = errStat == ErrID_Fatal
+      if (newInputMissing) then
+         call LegacyWarning(trim(varName)//' should be present on line '//trim(num2lstr(iLine))//'.')
+      else
+         ! We are happy
+      endif
+      ! We erase the error
+      errStat = ErrID_None
+      errMsg  = ''
+   end function newInputMissing
+
+   !-------------------------------------------------------------------------------------------------
+   subroutine printNewOldInputs()
+      ! Temporary HACK, for WakeMod=10, 11 or 12 use AeroProjMod 2 (will trigger PolarBEM)
+      if (InputFileData%Wake_Mod==10) then
+         call WrScr('[WARN] Wake_Mod=10 is a temporary hack. Setting BEM_Mod to 0')
+         InputFileData%BEM_Mod = 0
+      elseif (InputFileData%Wake_Mod==11) then
+         call WrScr('[WARN] Wake_Mod=11 is a temporary hack. Setting BEM_Mod to 2')
+         InputFileData%BEM_Mod = 2
+      elseif (InputFileData%Wake_Mod==12) then
+         call WrScr('[WARN] Wake_Mod=12 is a temporary hack. Setting BEM_Mod to 2')
+         InputFileData%BEM_Mod = 2
+      endif
+      !====== Summary of new AeroDyn options ===============================================================
+      ! NOTE: remove me in future release
+      call WrScr('-------------- New AeroDyn inputs (with new meaning):')
+      write (*,'(A20,I0)') 'Wake_Mod: '         , InputFileData%Wake_Mod
```

Comment:
Change `write (*,` to `write (CU,` for compatibility with Simulink

Reply:
I don't think `Write(CU,...)` works well in Simulink, either. `WrScr()` is better since that will call the Matlab `mexPrintf` function to write to the screen.

Reply:
Good point.  I had forgotten that `WrScr()` calls that.

Reply:
de83885ba